packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -247,7 +247,7 @@ export const useGridKeyboardNavigation = (
 
       const viewportPageSize = apiRef.current.unstable_getViewportPageSize();
       const colIndexBefore = params.field ? apiRef.current.getColumnIndex(params.field) : 0;
-      const firstRowIndexInPage = 0;
+      const firstRowIndexInPage = currentPageRows.length > 0 ? 0 : null;
```

Comment:
```suggestion
      const firstRowIndexInPage = currentPageRows.length > 0 ? 0 : -1;
```

Reply:
After doing this change you might update if conditions for cases `ArrowDown` and `PageDown` like so:

**ArrowDown:**
```diff
- if (firstRowIndexInPage !== null) {
+ if (firstRowIndexInPage >= 0) {
```

**PageDown:**
```diff
- if (firstRowIndexInPage !== null && lastRowIndexInPage !== null) {
+ if (firstRowIndexInPage >= 0 && lastRowIndexInPage >= 0) {
```

Reply:
@Dekshut Please do above mentioned changes for `handleColumnHeaderKeyDown` to make it work properly

Reply:
I guess I did. Please check it again